keps/582-preempt-based-on-flavor-order/kep.yaml

```diff
@@ -0,0 +1,31 @@
+title: KEP Template
+kep-number: 582
+authors:
+  - "@kunwuluan"
+status: provisional
+creation-date: 2023-05-17
+reviewers:
+  - "@alculquicondor"
+approvers:
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha|beta|stable
```

Comment:
I think the logic will be fairly simple, so I would be comfortable skipping alpha for this feature.